lib/config.js

```diff
@@ -369,17 +376,40 @@ function readEnv () {
 }
 
 function normalize (opts) {
-  normalizeIgnoreOptions(opts)
   normalizeKeyValuePairs(opts)
   normalizeNumbers(opts)
   normalizeBytes(opts)
   normalizeArrays(opts)
   normalizeTime(opts)
   normalizeBools(opts)
+  normalizeIgnoreOptions(opts)
   truncateOptions(opts)
 }
 
 function normalizeIgnoreOptions (opts) {
+  if (opts.transactionIgnoreUrls) {
+    // if users accidently configure a string instead of an
+    // array with a single string, handle it for them
+    if (!Array.isArray(opts.transactionIgnoreUrls)) {
+      opts.transactionIgnoreUrls = [opts.transactionIgnoreUrls]
+    }
+
+    // We can't guarentee that opts will be a Config so set a
+    // default value. This is to work around CENTRAL_CONFIG tests
+    // that call this method with a plain object `{}`
+    if (!opts.transactionIgnoreUrlRegExp) {
+      opts.transactionIgnoreUrlRegExp = []
+    }
+    const wildcard = new WildcardMatcher()
+    for (let ptn of opts.transactionIgnoreUrls) {
+      // if configured without a leading '/', be nice
+      // and add this to pattern for the user
+      if (ptn[0] !== '/') ptn = '/' + ptn
```

Comment:
I don't think we should do this "favour". 

- I suspect other agents don't do it, which would slightly/subtly break commonality. The Python agent at least doesn't do this.
- Combined with a possible empty string (which could come from a dangling comma like `ELASTIC_TRANSACTION_IGNORE_URLS=a,b,`) this would result in a pattern to ignore the root URL, `'/'`, which would be a surprise.

Reply:
That's fair -- I can back off this one.  (Likely came from years of trauma with apache handling the path differently in the main config vs. `.htaccess` files :))